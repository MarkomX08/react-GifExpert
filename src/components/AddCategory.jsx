import { useState } from "react"

export const AddCategory = ( { onNewCategory } ) => {

    const [InputValue, setInputValue] = useState('');

    // funcion: obtencion del valor puesto en el input
    const onInputChange =({ target })=> {
        setInputValue( target.value ); //obtencion del valor del input
    }

    // funcion: insercion del nuevo elemento en el array
    const onSubmit = ( event ) => {
        event.preventDefault();
        if( InputValue.trim().length <= 1) return;// condicion: verificacion del input, si viene vacio evita la insercion al array
        // setCategories(categories => [InputValue, ...categories]); //agregado del nuevo elemento al array
        onNewCategory( InputValue.trim() );
        setInputValue('');//limpieza del input
    }


    return (
        <form onSubmit={ onSubmit }>
            <input type="text" 
            placeholder="Buscar Gifs"
            value={ InputValue }
            onChange={ onInputChange }
            />
        </form>
    )
}
