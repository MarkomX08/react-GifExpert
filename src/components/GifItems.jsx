import propTypes from 'prop-types';
export const GifItems = ({title, url, id }) => {


    return (
        <div className="card">
            <img src={ url } alt={title} />
            <p> { title } </p>
        </div>
    )

}

GifItems.propTypes = {
    title: propTypes.string.isRequired,
    url: propTypes.string.isRequired
}
/* Tarea
1. AÃ±adir PrpTypes
    a. title obligatorio
    b. url obligatorio
2.Evaluar el snapshot

*/