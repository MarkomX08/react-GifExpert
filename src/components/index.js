export * from './AddCategory';
export * from './GifFrid';
export * from './GifItems';
