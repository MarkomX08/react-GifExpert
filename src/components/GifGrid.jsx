import PropTypes from 'prop-types';

import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItems } from "./GifItems";


export const GifGrid = ({ category }) => {

    const {images, isLoading} = useFetchGifs( category );


    return (
        <>
            <h3>{ category }</h3>
            
            {
                isLoading &&( <h2>Cargando...</h2>)
            }
            
            {/* card para los gifs */}
            <div className="card-grid">
                {
                    images.map( ( image ) =>(
                        <GifItems 
                            key={ image.id} 
                            { ...image }
                        />
                    ))
                }
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}
