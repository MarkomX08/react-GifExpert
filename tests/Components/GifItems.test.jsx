import { render,screen } from "@testing-library/react";
import { GifItems } from "../../src/components/GifItems";

describe('Pruebas en el componente de GifItems', () => { 

    const title= 'Dragon ball';
    const url = 'htpps://dragonball.com/goku.jpg';

    test('Inicializacion del snapshot', () => { 
        const { container } = render( <GifItems title= {title} url = {url} />);
        expect ( container ).toMatchSnapshot();
     });

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItems title= {title} url={url} /> );
        // expect( screen.getByRole('img').src).toBe( url );
        // expect( screen.getByRole('img').alt).toBe( title);
        
        const {src, alt} = screen.getByRole('img');
        expect( src ).toBe( url );
        expect( alt ).toBe( alt);
    });
    
    test('debe de mostrar el titulo en el componente', () => { 
        render(<GifItems title= {title} url={url} /> );
        expect( screen.getByText(title)).toBeTruthy();
      })
 })