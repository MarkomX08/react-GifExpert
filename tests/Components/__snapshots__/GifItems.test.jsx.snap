// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente de GifItems Inicializacion del snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="htpps://dragonball.com/goku.jpg"
    />
    <p>
       
      Dragon ball
       
    </p>
  </div>
</div>
`;
